### PR TITLE
[WIP] issue #1063 parse DA

### PR DESCRIPTION
**Pull request by victorjz**
- this is strictly for parsing and testing the district attorney number
- unsure whether to go with least invasive additions or fully robust
- case_parser new member as optional so as not to have to add arguments when calling the constructor throughout the project
- you'll see in the new parsing method __parse_distr_atty that there's potential to return all the values from upper right of a given case html file but i just return DA number
- would it be preferred to return more values? or return None instead of empty str?
- also added testing methods to test_case_parser

### CODE REVIEW

File: src/backend/expungeservice/crawler/parsers/case_parser.py [hunk 1]
```diff
@@ -37,8 +39,19 @@ def feed(data) -> CaseParserData:
...
+    @staticmethod
+    def __parse_distr_atty(soup) -> str:
+        distr_atty_key = "District Attorney Number:"
+        labels = soup.find_all("th", "ssTableHeaderLabel", limit=10)
+        table = {}
+        for tag in labels:
+            table[tag.string] = tag.parent.find("td").string
+        if distr_atty_key in table:
+            return table[distr_atty_key]
+        return ""
```
Comment (wittejm) [inline src/backend/expungeservice/crawler/parsers/case_parser.py:53]:
```suggestion
        else: 
            return ""
```

**Comment by wittejm** [comment]
Same thing with `CaseSummary` -- you can update all those constructor calls too.

**Comment by victorjz** [comment]
I think I submitted my new changes?  This is my first time modifying code in a pull request.

File: src/backend/expungeservice/crawler/parsers/case_parser.py [hunk 2]
```diff
@@ -37,8 +39,20 @@ def feed(data) -> CaseParserData:
...
+    def __parse_district_attorney_number(soup) -> str:
+        district_attorney_key = "District Attorney Number:"
+        labels = soup.find_all("th", "ssTableHeaderLabel", limit=10)
+        table = {}
+        for tag in labels:
+            table[tag.string] = tag.parent.find("td").string
+        if district_attorney_key in table:
+            return table[district_attorney_key]
+        else:
+            return ""
```
Comment (KentShikama) [inline src/backend/expungeservice/crawler/parsers/case_parser.py:54]:
```suggestion
        DISTRICT_ATTORNEY_KEY = "District Attorney Number:"
        labels = soup.find_all("th", "ssTableHeaderLabel", limit=10)
        table = {tag.string: tag.parent.find("td").string for tag in labels}
        return table.get(DISTRICT_ATTORNEY_KEY, "")
```
Note I have not tested this.

Comment (victorjz) [reply src/backend/expungeservice/crawler/parsers/case_parser.py:54]: I ran the proposed code changes in a local directory and it works fine.

I'm not familiar with how to proceed with changes. Do I click commit suggestion or do I test everything in docker again and do another pull request?

Comment (KentShikama) [reply src/backend/expungeservice/crawler/parsers/case_parser.py:54]: Clicking accept commit suggestion works. You can also just commit locally and push the commit (not a new PR).

**Comment by KentShikama** [comment]
Thanks @victorjz !

**Pull request merged**